src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs

```diff
@@ -145,7 +145,7 @@ private static async Task<Document> ConvertToEqualsAsync(Document document, Sema
                 false =>
                     generator.InvocationExpression(
                         generator.MemberAccessExpression(
-                            generator.TypeExpression(semanticModel.Compilation.GetSpecialType(SpecialType.System_Object)),
+                            generator.TypeExpression(SpecialType.System_Object),
```

Comment:
I guess that's a new API on roslyn side? I don't know why we would have gone through this more complex syntax otherwise.

Reply:
@Evangelink Both APIs were introduced in 3.0.0. I think this approach was taken since this is the first overload shown by IntelliSense?